lib/gpw/client.rb

```diff
@@ -27,11 +36,69 @@ def parse_response(response)
       body_parsed = JSON.parse(response.body.split('(')[1].split(')')[0])
 
       body_parsed.fetch('symbols').map do |company|
+        next unless company[1]
         {
           name: company[0],
           isin: company[1],
         }
       end
     end
+
+    def fetch_gpw_stock_list
+      conn = Faraday.new(
+        url: 'https://www.money.pl/api/graphql?query=query%20sg_money_gielda_gpw_stock_fundaments_compare(%24index%3A%20String%2C%20%24cp%3A%20Float%2C%20%24cz%3A%20Float%2C%20%24cwk%3A%20Float%2C%20%24wk_min%3A%20Float%2C%20%24wk_max%3A%20Float%2C%20%24kapitalizacja_min%3A%20Float%2C%20%24kapitalizacja_max%3A%20Float%2C%20%24ros%3A%20Float%2C%20%24roa%3A%20Float%2C%20%24roe%3A%20Float%2C%20%24zadluzenie_o%3A%20Float%2C%20%24zadluzenie_k%3A%20Float%2C%20%24zadluzenie_d%3A%20Float)%20%7B%0A%20%20tableData%3A%20gpw_stock_fundaments_compare(index%3A%20%24index%2C%20cp%3A%20%24cp%2C%20cz%3A%20%24cz%2C%20cwk%3A%20%24cwk%2C%20wk_min%3A%20%24wk_min%2C%20wk_max%3A%20%24wk_max%2C%20kapitalizacja_min%3A%20%24kapitalizacja_min%2C%20kapitalizacja_max%3A%20%24kapitalizacja_max%2C%20ros%3A%20%24ros%2C%20roa%3A%20%24roa%2C%20roe%3A%20%24roe%2C%20zadluzenie_o%3A%20%24zadluzenie_o%2C%20zadluzenie_k%3A%20%24zadluzenie_k%2C%20zadluzenie_d%3A%20%24zadluzenie_d)%20%7B%0A%20%20%20%20stock%0A%20%20%20%20stock_url_rel%0A%20%20%20%20cp%0A%20%20%20%20cz%0A%20%20%20%20cwk%0A%20%20%20%20wk%0A%20%20%20%20kapitalizacja%0A%20%20%20%20roa%0A%20%20%20%20roe%0A%20%20%20%20ros%0A%20%20%20%20zadluzenie_o%0A%20%20%20%20zadluzenie_k%0A%20%20%20%20zadluzenie_d%0A%20%20%20%20report_date%0A%20%20%20%20stock_url_rel%0A%20%20%20%20__typename%0A%20%20%7D%0A%7D%0A&operationName=sg_money_gielda_gpw_stock_fundaments_compare&variables=%7B%7D',
+        headers: {
+          'Accept' => '*/*',
+          'DNT' => '1',
+          'Referer' => 'https://www.money.pl',
+        },
+      )
+
+      conn.get
+    end
+
+    def parse_money_response_gpw(response)
+      response_parsed = JSON.parse(response.body)
+
+      response_parsed['data']['tableData'].map do |company|
+        next unless company['stock_url_rel']
```

Comment:
My proposal:

```suggestion
      response_parsed
        .fetch('data')
        .fetch('tableData')
        .select { |row| row['stock_url_rel'] }
        .map do |company|
```

Then you would probably not need to use `compact!` later.